### ISSUE TITLE
Truncating reads to 1

### ISSUE THREAD

**Issue by robinsleith** [BigelowLab/edna-dada2#30]
Miss me yet?

Just started with a new dataset and with truncLen auto I am getting reverse reads truncLen set to 1 and losing all reads. Is this an issue with the line intersecting at two points? Troubleshooting read examples are here /mnt/storage/data/edna/dada/projects/robin_foo/mann/

quality profiles for all attached but see aug-1 and feb-4 for the reads I isolated

truncLen.csv output:
```
forward,forward_file,reverse,reverse_file
268,/mnt/storage/data/edna/dada/projects/robin_foo/mann/process/cutadapt/aug-1_S114_L001_R1_001.fastq.gz,1,/mnt/storage/data/edna/dada/projects/robin_foo/mann/process/cutadapt/aug-1_S114_L001_R2_001.fastq.gz
276,/mnt/storage/data/edna/dada/projects/robin_foo/mann/process/cutadapt/feb-4_S101_L001_R1_001.fastq.gz,199,/mnt/storage/data/edna/dada/projects/robin_foo/mann/process/cutadapt/feb-4_S101_L001_R2_001.fastq.gz
```
filterandtrim.csv output:
```
name,reads.in,reads.out
aug-1_S114_L001_R1_001.fastq.gz,28333,0
feb-4_S101_L001_R1_001.fastq.gz,7407,6311
```
[quality_profiles.pdf](https://github.com/BigelowLab/edna-dada2/files/11082308/quality_profiles.pdf)

**Comment by btupper** [comment]
Oh, geez.  They don't look very different from the other ones.

**Comment by btupper** [comment]
I'm not finding you input data.  Might we make a subsample for just those two datasets?

```
btupper@cfe1 ~ $ cd $EDNA/dada/projects/robin_foo/mann
btupper@cfe1 mann $ ll
total 60
-rw-rw-r-- 1 rsleith edna  7979 Mar 27 14:16 asv_18S_postprocess.R
-rw-rw-r-- 1 rsleith edna  1275 Mar 27 14:16 asv_18S_postprocess.yaml
-rw-rw-r-- 1 rsleith edna  9491 Mar 27 14:16 asv_18S_preprocess.R
-rw-rw-r-- 1 rsleith edna  1752 Mar 27 14:53 asv_18S_preprocess.yaml
drwxrwsr-x 4 rsleith edna  4096 Aug 16  2022 aug16_process
drwxrwsr-x 4 rsleith edna  4096 Mar 27 15:21 auto_process
-rw-rw-r-- 1 rsleith edna   393 Aug 16  2022 job_submission.sh
drwxrwsr-x 4 rsleith edna  4096 Jul  5  2022 jul5_process
drwxrwsr-x 3 rsleith edna 16384 Mar 27 15:28 raw_reads
```

**Comment by btupper** [comment]
Never mind - you put them in raw_reads.  Got it.

**Comment by robinsleith** [comment]
Any time this week to look this over?

**Comment by btupper** [comment]
Sure - how does tomorrow (Tues) morning look?

**Comment by btupper** [comment]
Works for me

**Comment by robinsleith** [comment]
I think I understand why we implemented "too few below min_fraction_above_threshold - cutoff assigned 1" and it was to deal with low quality reads and weirdly shaped functions, but these reads are pretty normal so I am stuck trying to figure out why it is assigning a cutoff of 1. Is there a way to look at the min_fraction_above_threshold values for these 2 sets of reads? I am around until 11 this morning and happy to chat!

**Comment by btupper** [comment]
Added min_threshold and verbosity to cutoff_params in YAML

**Comment by robinsleith** [comment]
I am having an issue I think related to this verbosity change. I am guessing it is a mismatch between R script and yaml. Working here /mnt/storage/data/edna/dada/projects/robin_foo/julia

```
Error in (function (x1, key, threshold = 30, quantile_min = 0.9, model = stats::as.formula("Mean ~ poly(Cycle, 2)"),  : 
  unused argument (verbose = FALSE)
```

**Comment by btupper** [comment]
That certainly looks like the culprit.  I'll take a look.

**Comment by robinsleith** [comment]
I am going to try commenting out this section https://github.com/BigelowLab/dadautils/blob/0c20ba92bafb1d157074db3676439581bee3f7af/R/quality_profile.R#L87

**Comment by robinsleith** [comment]
Welp, that didnt work, I changed things back. Ill be out for the next 1.5 hour but will take another look later.

**Comment by btupper** [comment]
Let's walk through a reinstall for you before we go much further.  Reinstall dadautils, charlier and auntie.  Use `/mnt/storage/data/edna/packages/edna_user_installer.R` perhaps?

**Comment by robinsleith** [comment]
I have been running this line every time I'm trouble shooting and changing something. I am running it correctly?

Rscript /mnt/storage/data/edna/packages/edna_user_installer.R charlier auntie dadautils

**Comment by btupper** [comment]
And you have /mod/Bigelow/dada2 exposed?  Not some other instance of R?

**Comment by robinsleith** [comment]
Yeah by purge and then load module dada2. But its possible there something going on with updates on head node vs execute node. I can troubleshoot that using an interactive session.

**Comment by btupper** [comment]
Oh - found it!  I added `verbose = FALSE` to the argument list here... https://github.com/BigelowLab/dadautils/blob/main/R/quality_profile.R#L110

Reinstall and try ~banging your head against the wall~ again?

**Comment by robinsleith** [comment]
Awesome, I will let you know how it goes. Thanks!

**Comment by robinsleith** [comment]
Is there an errant comma?
```
Error in parse(outFile) : 
  /var/tmp/pbs.1547304.cfe1/RtmpJ1lB1F/R.INSTALL1029b15f5d13f2/dadautils/R/quality_profile.R:86:34: unexpected ','
85:   params = list(score = 30, model = "Mean ~ poly(Cycle, 2)", quantile_min = 0.9, min_fraction_above_threshold = 0.7, verbose = TRUE),
86:   form = c("full", "reduced")[2]),
                                     ^
ERROR: unable to collate and parse R files for package ‘dadautils’
```

**Comment by btupper** [comment]
Yes - but an errant `)` from maybe when you tried commenting out the verbose?  Fixed - please try ~sticking fingers in fan~ again.

**Comment by robinsleith** [comment]
It worked! Thanks, closing this issue!
